src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -181,12 +183,15 @@ public static object Construct(Type type, object[] arguments)
         /// </summary>
         /// <param name="objects">An array of objects</param>
         /// <returns>An array of Types</returns>
-        private static Type[] GetTypeArray(object[] objects)
+        internal static Type[] GetTypeArray(object[] objects)
         {
             Type[] types = new Type[objects.Length];
             int index = 0;
             foreach (object o in objects)
-                types[index++] = o.GetType();
+            {
+                // NUnitNullType is a marker to indicate null since we can't do typeof(null) or null.GetType()
+                types[index++] = o == null ? typeof(NUnitNullType) : o.GetType();
```

Comment:
Even if it is a typed null, in this case it still come in as a null parameter. Just to be sure, I added a typed null test and it works fine.